tests/c/fcmp_double.newcg.c

```diff
@@ -0,0 +1,85 @@
+// ignore-if: test $YK_JIT_COMPILER != "yk" -o "$YKB_TRACER" = "swt"
+// Run-time:
+//   env-var: YKD_LOG_IR=-:aot,jit-pre-opt,jit-asm
+//   env-var: YKD_SERIALISE_COMPILATION=1
+//   env-var: YKD_LOG_JITSTATE=-
+//   stderr:
+//     jitstate: start-tracing
+//     1.100000 == 2.200000: 0
+//     1.100000 == 1.100000: 1
+//     1.100000 != 2.200000: 1
+//     1.100000 > 2.200000: 0
+//     1.100000 < 2.200000: 1
+//     1.100000 >= 2.200000: 0
+//     1.100000 <= 2.200000: 1
+//     1.100000 >= 1.100000: 1
+//     1.100000 <= 1.100000: 1
+//     1.100000 <= nan: 0
+//     nan == nan: 0
+//     nan != nan: 1
+//     0.000000 == -0.000000: 1
+//     jitstate: stop-tracing
+//     ...
+//     jitstate: enter-jit-code
+//     1.100000 == 2.200000: 0
+//     1.100000 == 1.100000: 1
+//     1.100000 != 2.200000: 1
+//     1.100000 > 2.200000: 0
+//     1.100000 < 2.200000: 1
+//     1.100000 >= 2.200000: 0
+//     1.100000 <= 2.200000: 1
+//     1.100000 >= 1.100000: 1
+//     1.100000 <= 1.100000: 1
+//     1.100000 <= nan: 0
+//     nan == nan: 0
+//     nan != nan: 1
+//     0.000000 == -0.000000: 1
+//     jitstate: deoptimise
+
+// Check that double comparisons work.
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk.h>
+#include <yk_testing.h>
+
+int main(int argc, char **argv) {
+  YkMT *mt = yk_mt_new(NULL);
+  yk_mt_hot_threshold_set(mt, 0);
+  YkLocation loc = yk_location_new();
+
+  int i = 3;
+  double a = 1.1, b = 2.2, nana = NAN;
+  double aa = a, nanb = nana;
+  double zero = 0.0, negzero = -0.0;
+  NOOPT_VAL(loc);
+  NOOPT_VAL(i);
+  NOOPT_VAL(a);
+  NOOPT_VAL(aa);
+  NOOPT_VAL(b);
+  NOOPT_VAL(nana);
+  NOOPT_VAL(nanb);
```

Comment:
Do we need to `NOOPT_VAL` 'zero' and 'negzero'?

Reply:
Yes, we do. Well spotted.